pkg/controller/plan/kubevirt.go

```diff
@@ -327,17 +326,10 @@ func (r *KubeVirt) EnsureVM(vm *plan.VMStatus) (err error) {
 
 	for _, pvc := range pvcs {
 		ownerRefs := []meta.OwnerReference{vmOwnerReference(virtualMachine)}
-		patch := map[string]interface{}{
-			"metadata": map[string]interface{}{
-				"ownerReferences": ownerRefs,
-			},
-		}
-		patchBytes, patchErr := json.Marshal(patch)
-		if patchErr != nil {
-			err = liberr.Wrap(patchErr)
-			return
-		}
-		err = r.Client.Patch(context.TODO(), &pvc, client.RawPatch(types.MergePatchType, patchBytes))
+		pvcCopy := pvc.DeepCopy()
+		pvcCopy.OwnerReferences = ownerRefs
+		patch := client.MergeFrom(pvcCopy)
+		err = r.Client.Patch(context.TODO(), &pvc, patch)
```

Comment:
I like the idea of using MergeFrom but looking at the tests [here](https://github.com/kubernetes-sigs/controller-runtime/blob/0ef07535542bb44dc642fe5c67dcd4f7a9c06a22/pkg/client/fake/client_test.go), shouldn't it be 
```golang
pvcCopy := pvc.DeepCopy()
pvc.OwnerReferences = ownerRefs // update `pvc` here
patch := client.MergeFrom(pvcCopy)
err = r.Client.Patch(context.TODO(), &pvc, patch)
```

Reply:
Indeed, also in https://github.com/kubev2v/forklift/blob/5c358f5a71e11f9dde4a13e49c06ffc415ea501a/pkg/controller/plan/kubevirt.go#L459